core/src/main/scala/cats/Monad.scala

```diff
@@ -13,10 +13,16 @@ import scala.annotation.implicitNotFound
  * Must obey the laws defined in cats.laws.MonadLaws.
  */
 @implicitNotFound("Could not find an instance of Monad for ${F}")
-@typeclass trait Monad[F[_]] extends FlatMap[F] with Applicative[F] {
+@typeclass trait Monad[F[_]] extends FlatMap[F] with Selective[F] {
   override def map[A, B](fa: F[A])(f: A => B): F[B] =
     flatMap(fa)(a => pure(f(a)))
 
+  override def select[A, B](fab: F[Either[A, B]])(ff: F[A => B]): F[B] =
```

Comment:
An implementation equivalent to this is the law in Haskell for monads.  Left open to overrides for efficiency.

Reply:
`FlatMap` should extend `Select`, but once it does, we need to push this down.  And once we do, we no longer have `pure` and lose our default implementation.

I think we can salvage `Selective`, but not `Select`.

Reply:
Well, we still can do it from `apply`.  Maybe it lives.